### PR TITLE
DVO Validation Updates

### PR DESCRIPTION
**Pull request by TGPSKI**
This PR attempts to fix DVO validation errors for openshift-acme resources. Not all of the validations are fixable, and careful thought must be applied to consider if the applied fixes are valuable or not. I am looking for feedback and the team's thoughts on how to proceed with these changes. 

| check  | notes |
|---|---|
|  deprecated_service_account_field | serviceAccountName is replaced with serviceAccount at the kube-api level. oc process does not mutate the serviceAccountName key, so the mutation must be in kube itself. @bkez322's pr to kube-linter should resolve this issue. |
| minimum_three_replicas  |  This check should not apply for this deployment. Only a single replica is necessary to provide the controller functionality.  |
| no_liveness_probe  | This is a point of debate for me. Liveness and readiness probes should be implemented at the controller level and exposed to those implementing the controller such that they can add it to their manifests. The check I added looks for the presence of the controller executable, but can not tell if the controller is working. I debate whether or not this probe is useful, and want to open it up for discussion. |
|  non_existent_service_account | Failing this check is a side-effect of statically analyzing k8s objects running on a live cluster without considering dependent objects..  |
| no_readiness_probe | Same debate as "no_liveness_probe" |
| no_read_only_root_fs | SCC is an OpenShift concept - workloads are bound by SCC's, and kube-linter does not have knowledge of SCCs. Given workloads on OpenShift are bound by SCC, I do not think this check is valuable. |
| required_annotation_email | Straightforward to implement and add, but requires ensuring that all dependent deployments of openshift-acme-controller have the required fields applied in their respective saas files. |
| required_label_owner | Same as email. |

### CODE REVIEW

File: openshift/openshift-acme.template.yaml [hunk 1]
```diff
@@ -120,6 +123,22 @@ objects:
             limits:
               cpu: ${CPU_LIMITS}
               memory: ${MEMORY_LIMITS}
+          livenessProbe:
+            exec:
+              command:
+              - /bin/sh
+              - -c
+              - "ps -A | grep openshift-acme"
```
Comment (Piojo) [inline openshift/openshift-acme.template.yaml:131]: If all you care about is the existence of the process, I'd rather go with `pkill -0 openshift-acme` or `killall -0 openshift-acme`. If you have `ps` you should have `pkill`.

Comment (TGPSKI) [reply openshift/openshift-acme.template.yaml:131]: pkill confirmed

Comment (TGPSKI) [reply openshift/openshift-acme.template.yaml:131]: I care more than just the existence of the process, but there is nothing exposed in the controller that provides any liveness or readiness checks. This is my probem - are these checks even valuable?

Comment (TGPSKI) [reply openshift/openshift-acme.template.yaml:131]: The natural route here would be to go upstream and get some kind of liveness/readiness added to the controller. 

Question is probably for @rrati: do we prioritize cleaning as many failures as possible, or doing them properly? Properly means going upstream and waiting for that cycle to complete, which may not match with deliverable timelines. Cleaning fast means adding these checks temporarily while we wait for upstream changes.

Comment (rrati) [reply openshift/openshift-acme.template.yaml:131]: The goal of this is to improve our services and to validate DVO's checks.  This is a valid check, so the solution should be correct as well.  Unless we can avoid it we should always do the fix properly.

We have the ability to skip specific checks if needed, but if that is done we should log a ticket for how to fix this properly.

**Comment by TGPSKI** [comment]
I'm closing this for the time being. I have a ticket to track this work and we'll need to get readiness/liveness added upstream. Since this touches a ton of services, definitely need to do it the right way and not worry about speed as much.

**Pull request closed without merging**